src/identity/mod.rs

```diff
@@ -0,0 +1,281 @@
+                let new_id = Self::new()?;
+                match new_id.save() {
+                    Ok(()) => Ok(new_id),
+                    Err(err) => Err(err),
+                }
+            }
+            Err(err) => Err(err),
+        }
+    }
+
+    /// Convert public key to a base64 encoded string
+    /// Encodes public key as DER and then encodes DER into base64
+    pub fn public_key_to_base64(&self) -> Result<String, AetherError> {
+        let public_key_der = self.rsa.public_key_to_der()?;
+        Ok(base64::encode(public_key_der))
+    }
+
+    /// Convert private key to a base64 encoded string
+    /// Encodes private key as DER and then encodes DER into base64
+    pub fn private_key_to_base64(&self) -> Result<String, AetherError> {
+        let private_key_der = self.rsa.private_key_to_der()?;
+        Ok(base64::encode(private_key_der))
+    }
+
+    /// Encrypt given bytes using the public key
+    pub fn public_encrypt(&self, from: &[u8]) -> Result<Vec<u8>, AetherError> {
+        let mut buf: Vec<u8> = vec![0; self.rsa.size() as usize];
+        self.rsa.public_encrypt(from, &mut buf, Padding::PKCS1)?;
+        Ok(buf.to_vec())
+    }
+
+    /// Encrypt given bytes using the private key
+    pub fn private_encrypt(&self, from: &[u8]) -> Result<Vec<u8>, AetherError> {
+        let mut buf: Vec<u8> = vec![0; self.rsa.size() as usize];
+        self.rsa.private_encrypt(from, &mut buf, Padding::PKCS1)?;
+        Ok(buf.to_vec())
+    }
+
+    /// Decrypt given bytes using the public key
+    pub fn public_decrypt(&self, from: &[u8]) -> Result<Vec<u8>, AetherError> {
+        let mut buf: Vec<u8> = vec![0; self.rsa.size() as usize];
+        let size = self.rsa.public_decrypt(from, &mut buf, Padding::PKCS1)?;
+        Ok(buf[..size].to_vec())
+    }
+
+    /// Decrypt given bytes using the private key
+    pub fn private_decrypt(&self, from: &[u8]) -> Result<Vec<u8>, AetherError> {
+        let mut buf: Vec<u8> = vec![0; self.rsa.size() as usize];
+        let size = self.rsa.private_decrypt(from, &mut buf, Padding::PKCS1)?;
+        Ok(buf[..size].to_vec())
+    }
+}
+
+impl PublicId {
+    /// Decode the given base64 string into a [`PublicId`]
+    /// # Errors
+    /// * [`AetherError::Base64DecodeError`]    -   If the given string is not valid base64
+    pub fn from_base64(key: &str) -> Result<PublicId, AetherError> {
+        let bytes = base64::decode(key)?;
+        let rsa = Rsa::public_key_from_der(&bytes)?;
+        Ok(Self { rsa })
+    }
+
+    /// Convert public key to a base64 encoded string
+    /// Encodes public key as DER and then encodes DER into base64
+    pub fn public_key_to_base64(&self) -> Result<String, AetherError> {
+        let public_key_der = self.rsa.public_key_to_der()?;
+        Ok(base64::encode(public_key_der))
+    }
+
+    /// Encrypt given bytes using the public key
+    pub fn public_encrypt(&self, from: &[u8]) -> Result<Vec<u8>, AetherError> {
```

Comment:
This could be a utility function as both use the same right? Or are you doing it like this for any security reasons

Reply:
Even if we define it as a trait instead, there will have to be 2 implementation anyways for `Id` and `PublicId`